app/src/main/java/chat/rocket/android/sharehadler/ShareHandler.kt

```diff
@@ -0,0 +1,136 @@
+package chat.rocket.android.sharehadler
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Parcelable
+import android.provider.OpenableColumns
+import java.io.InputStream
+
+object ShareHandler {
+
+    fun hasShare(): Boolean = hasSharedText() || hasSharedFile()
+
+    fun hasSharedText(): Boolean = sharedText != null
+    fun hasSharedFile(): Boolean = files.size > 0
+
+    var sharedText: String? = null
+
+    var files: ArrayList<SharedFile> = arrayListOf()
+
+    fun handle(intent: Intent?, context: Context) {
+        clearAll()
+
+        intent?.let {
+
+            val action = it.action
+            val type = it.type
+
+            type?.let {
+                when (action) {
+                    Intent.ACTION_SEND -> {
+                        when {
+                            "text/plain" == type -> {
+                                // Handle text being sent
+                                handleSendText(intent)
+                            }
+                            type.startsWith("image/") -> {
```

Comment:
Is "startsWith("image/") neccessary, since it's the same as action as in the else clause?  Can't you just use the else clause here?